e3sm_diags/driver/utils/regrid.py

```diff
@@ -383,20 +389,62 @@ def align_grids_to_lower_res(
 
     if is_a_lower_res:
         output_grid = ds_a_new.regridder.grid
+        # Only create mask for 2D data (with no vertical dimension [zonal_mean_2d_*, meridional_mean_2d], or time dimension [annual_cycle_zonal_mean])
+
+        ds_b_new = _add_mask_for_2d(
+            ds_b_new, var_key, tool, ["lev", "plev", "z", "Z", "time", "T"]
+        )
         ds_b_regrid = ds_b_new.regridder.horizontal(
             var_key, output_grid, tool=tool, method=method
         )
-
         return ds_a_new, ds_b_regrid
 
     output_grid = ds_b_new.regridder.grid
+    ds_a_new = _add_mask_for_2d(
+        ds_a_new, var_key, tool, ["lev", "plev", "z", "Z", "time", "T"]
+    )
+
     ds_a_regrid = ds_a_new.regridder.horizontal(
         var_key, output_grid, tool=tool, method=method
     )
 
     return ds_a_regrid, ds_b_new
 
 
+def _add_mask_for_2d(
+    ds: xr.Dataset, var_key: str, tool: str, dims_to_check: List[str]
+) -> xr.Dataset:
+    """Add a mask variable to the dataset if needed based on the tool and dimensions.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        The dataset to modify.
+    var_key : str
+        The variable key.
+    tool : str
+        The regridding tool being used.
+    dims_to_check : List[str]
+        A list of dimensions to check for skipping mask creation.
+
+    Returns
+    -------
+    xr.Dataset
+        The modified dataset with the mask variable added if applicable.
+    """
+    var_dims = ds[var_key].dims
+
+    if tool == "regrid2" or not any(dim in dims_to_check for dim in var_dims):
```

Comment:
```suggestion
    # Create a mask for xESMF if the variable is 2D (does not include specified dimensions)
    if tool == "xesmf" and not any(dim in dims_to_check for dim in var_dims):
```

Are we supposed to add the `mask` variable whenever the tool is `xesmf`, rather than `regrid2`?

Reply:
I think the logic is right without changing, assumming `regrid2` can take 3d var as input for regriding?